src/layers/components/BasketModal/BasketModal.ts

```diff
@@ -0,0 +1,357 @@
+        if (cardCVVInput) {
+            cardCVVInput.addEventListener('input', (e: Event) => {
+                if (e.target instanceof HTMLInputElement) {
+                    const value = e.target.value;
+                    const arr = value.split(' ');
+                    if (Number.isNaN(Number(arr[arr.length - 1]))) {
+                        cardCVVInput.value = `${value.slice(0, -1)}`;
+                    }
+
+                    if (value.length === 4) {
+                        cardCVVInput.value = `${value.slice(0, -1)}`;
+                    }
+                }
+            });
+        }
+    }
+
+    private isNameValid(str: string): boolean {
+        let result = true;
+        const arr = str.trim().split(' ');
+        if (arr.length < 2) {
+            return false;
+        }
+        arr.forEach((el) => {
+            if (el.trim().length < 3) {
+                result = false;
+            }
+            if (!/[a-zA-Z]{3}/g.test(el.trim())) {
+                result = false;
+            }
+        });
+
+        return result;
+    }
+
+    private isPhoneValid(str: string): boolean {
+        if (/\s/gm.test(str)) {
+            return false;
+        }
+        // eslint-disable-next-line no-useless-escape
+        if (/^((\+))(\(?\d{3}\)?[\- ]?)?[\d\- ]{9,}$/g.test(str)) {
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    private isAddressValid(str: string): boolean {
+        let result = true;
+        const arr = str.trim().split(' ');
+        if (arr.length < 3) {
+            return false;
+        }
+        arr.forEach((el) => {
+            if (el.trim().length < 5) {
+                result = false;
+            }
+        });
+
+        return result;
+    }
+
+    private isEmailValid(str: string): boolean {
+        if (/.+@.+\..+/g.test(str)) {
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    private isCardNumberValid(str: string): boolean {
+        let result = true;
+        const arr = str.trim().split(' ');
+        arr.forEach((el) => {
+            if (el.trim().length !== 4) {
+                result = false;
+            }
+            if (Number.isNaN(Number(el))) {
+                result = false;
+            }
+        });
+        if (str.trim().length !== 19) {
+            result = false;
+            return result;
+        } else {
+            return result;
+        }
+    }
+
+    private dateCardValid(str: string): boolean {
+        const arr = str.trim().split('/');
+        if (
+            arr[0].trim().length !== 2 ||
+            Number.isNaN(Number(arr[0])) ||
+            Number(arr[0].trim()) > 12 ||
+            arr[1].trim().length !== 2 ||
+            Number.isNaN(Number(arr[1]))
+        ) {
+            return false;
+        } else {
+            return true;
+        }
+    }
+
+    private isCCVValid(str: string): boolean {
```

Comment:
This function can be much simpler. At least you can delete excess variable result and return boolean without saving

Reply:
Update [396e7f4](https://github.com/llallonen/online-store/pull/65/commits/396e7f449f839d28275a9d259a41ece8e8d2e9d0)

Reply:
Update